thruster-core/src/macros.rs

```diff
@@ -0,0 +1,13 @@
+#[macro_export]
+macro_rules! map_try {
+  [ $expr:expr, $pat:pat => $mapper:expr ] => ({
+    match $expr {
+        Ok(val) => val,
+        $pat => {
+            let __e = $mapper;
+
+            return Err(__e);
```

Comment:
It might be slightly nicer to call `into` on the error similar to `?`.

```rust
return Error(Into::into(__e));
```

Reply:
Will include this in the next release